Update documentation to include write examples (#100)
georust/gpx: README.md

}
```

### Generate a new GPX file
This example only generates tracks. You can add waypoints and routes as well by instantiating new ``Waypoint``s and ``Route``s.
```rust
use std::path::Path;
use gpx::{Gpx, Track, TrackSegment, Waypoint, Route};
use geo_types::{Point, coord};

> pub fn to_gpx<P: AsRef<Path>>(out_path: P) -> Result<String, Box<dyn Error>> {

Reviewer: ```suggestion
pub fn to_gpx<P: AsRef<Path>>(out_path: P) -> Result<(), Box<dyn Error>> {
```

    // Instantiate Gpx struct

>     let mut gpx: Gpx = Gpx {

Reviewer: Let's build this in the other order (`let track_segment = ...; let track = ...; let gpx = ...`). That way wou don't need indexing and can do directly e.g. `vec![track]`.

>         version: GpxVersion::Gpx11, // or Gpx10

Reviewer: I'm personally not a fan of these type annotations / comments here. If you're just skimming the README, it's enough to know you can set e.g. the `waypoints`, but the exact type won't matter too much. And if you've pasted the code into your editor, you'll hopefully get enough info from it.

        creator: None, // Option<String>
        metadata: None, // Option<Metadata>
        waypoints: vec![], // Vec<Waypoint>
        tracks: vec![], // Vec<Track>
        routes: vec![], // Vec<Route>
    };
    gpx.tracks.push(Track {
        name: Some("Track 1".to_string()), // Option<String>
        comment: None, // Option<String>
        description: None, // Option<String>
        source: None, // Option<String>
        links: vec![], // Vec<Link>
        type_: None, // Option<String>
        number: None, // Option<u32>
        segments: vec![], // Vec<TrackSegment>
    });
    gpx.tracks[0].segments.push(TrackSegment { 
        points: vec![] // Vec<Waypoint>
    });

    // Create file at path

>     let gpx_file = File::create(out_path);

Reviewer: ```suggestion
    let gpx_file = File::create(out_path)?;
```

    
    // Add track point
    let geo_coord = coord! { x: -121.1, y: 38.82 };
    let geo_point: Point = geo_coord.into();
    gpx.tracks[0].segments[0].points.push(Waypoint::new(geo_point));

    // Write to file

>     write(&gpx, gpx_file?)?;

Reviewer: ```suggestion
    write(&gpx, gpx_file)?;
```

    Ok(());
}
```

### Write to string

> `write` will write the GPX output to anything that implements `std::io::Write`. To save the output to a string, use a mutable `BufWriter` to write it to a vector, and then convert the vector to a string.

Reviewer: I don't think you need to wrap a `Vec` in a `BufWriter`. But it might be a good idea to do it for files, since `write` might cause very short writes. So can you update both examples?

```rust
let mut buf = BufWriter::new(Vec::new());

> write(&gpx, &mut buf)?; // writes to buf

Reviewer: ```suggestion
gpx::write(&gpx, &mut buf)?; // writes to buf
```

}
```

### Generate a new GPX file
This example only generates tracks. You can add waypoints and routes as well by instantiating new ``Waypoint``s and ``Route``s.

```rust
use std::path::Path;
use gpx::{Gpx, Track, TrackSegment, Waypoint, Route};

> use geo_types::{Point, coord};

Reviewer: You're missing a couple of imports:

```rust
use geo_types::{coord, Point};
use gpx::{Gpx, GpxVersion, Track, TrackSegment, Waypoint};
use std::{error::Error, fs::File, io::BufWriter, path::Path};
```

pub fn to_gpx<P: AsRef<Path>>(out_path: P) -> Result<(), Box<dyn Error>> {
    // Instantiate Gpx struct
    let track_segment = TrackSegment {
        points: vec![]
    };
    let track = Track {
        name: Some("Track 1".to_string()),
        comment: None,
        description: None,
        source: None,
        links: vec![],
        type_: None,
        number: None,
        segments: vec![track_segment],
    };
    let mut gpx = Gpx {
        version: GpxVersion::Gpx11,
        creator: None,
        metadata: None,
        waypoints: vec![],
        tracks: vec![track],
        routes: vec![],
    };

    // Create file at path
    let gpx_file = File::create(out_path)?;
    let buf = BufWriter::new(gpx_file);

    // Add track point
    let geo_coord = coord! { x: -121.1, y: 38.82 };
    let geo_point: Point = geo_coord.into();
    gpx.tracks[0].segments[0].points.push(Waypoint::new(geo_point));

    // Write to file
    gpx::write(&gpx, buf)?;

>     Ok(());

Reviewer: ```suggestion
    Ok(())
```

}
```

### Write to string

> `write` will write the GPX output to anything that implements `std::io::Write`. To save the output to a string, write it to a UTF-8 vector, and then convert the vector to a string.

Reviewer: ```suggestion
`write` will write the GPX output to anything that implements `std::io::Write`. To save the output to a string, write it to an `u8` vector, and then convert the vector to a string.
```

```rust

> let mut vec: Vec<u8> = Vec::new();

Reviewer: ```suggestion
let mut vec = Vec::new();
```
